everestrs/everestrs_sys/everestrs_sys.cpp

```diff
@@ -57,13 +57,34 @@ void Module::signal_ready(const Runtime& rt) const {
     handle_->signal_ready();
 }
 
-void Module::provide_command(const Runtime& rt, const CommandMeta& meta) const {
-    handle_->provide_cmd(std::string(meta.implementation_id), std::string(meta.name), [&rt, meta](json args) {
-        JsonBlob blob = rt.handle_command(meta, json2blob(args));
+void Module::provide_command(const Runtime& rt, rust::String implementation_id, rust::String name) const {
+    handle_->provide_cmd(std::string(implementation_id), std::string(name), [&rt, implementation_id, name](json args) {
+        JsonBlob blob = rt.handle_command(implementation_id, name, json2blob(args));
         return json::parse(blob.data.begin(), blob.data.end());
     });
 }
 
+void Module::subscribe_variable(const Runtime& rt, rust::String implementation_id, rust::String name) const {
+    // TODO(hrapp): I am not sure how to model the multiple slots that could theoretically be here.
```

Comment:
@SirVer : what do you mean by multiple slots?

Reply:
@a-w50 I mean that `num_connections  != 1` for this `implementation_id`. I did not understand the C++ code around this, hence I did not model it in Rust at all currently.  My understanding is that the c++ code initially assumed that there would only be one connection, but then this was extendended to allow for more than one connection. To me this is a topic that needs addressing, but can be done at a later point. I also need a clinic where somebody explains this to me (or @dorezyuk, whoever implements this for rust).